### DISCUSSION TITLE
predictions with PCMCI

### DISCUSSION THREAD

**Discussion by AGraciaSipan** [jakobrunge/tigramite#220] [Q&A] [answered]
Hi, I have some doubts with the prediction tutorial. Is there a way to predict with the predictors discoverd when running pcmci algorithm? Because in the tutorial and in the documentation says that `get_predictors()` skips the MCI step. 

I've tried with `pcmci.all_parents` but doesn't return contemporaneous links and since I run `get_corrected_pvalues(fdr_method='fdr_bh')` there are more links than I want. 

And the `fit()` function in the documentantion says that "`target_predictors `(dictionary) – Dictionary of form {0:[(0, -1), (3, -2), …], 1:[], …} containing the predictors estimated with PCMCI." so I guess it is possible to get the estimated predictoris from PCMCI but I can't find how

**Comment by jvicentem** [comment]
Hi, I've also stumbled upon this. This was my solution:

```
def selected_links_from_pcmci(series_diff, results, alpha, target_index):
    selected_links = dict()

    mask = results['p_matrix'] <= alpha

    for i in range(0, series_diff.shape[1]):        
        for j in range(0, series_diff.shape[1]):
            if j not in selected_links:
                selected_links[j] = []
            for t, v in enumerate(mask[i,j]):
                if t > 0:
                    if v:
                        if i == target_index:
                            if i == j:
                                selected_links[j].append(tuple([i, -t]))
                        else:
                            selected_links[j].append(tuple([i, -t]))
                            
    return selected_links
```

```
 pred = Prediction(dataframe=tgm_df,
            cond_ind_test=parcorr,
            prediction_model = sklearn.linear_model.LinearRegression(),
        data_transform=None,
        train_indices= range(int(0.8*T)),
        test_indices= range(int(0.8*T), T),
        verbosity=1
        ) 

results = pred.run_pcmci(pc_alpha=[alpha], alpha_level=alpha, tau_max=tau_max)

selected_links = selected_links_from_pcmci(series_diff, results, alpha, target)

fit = pred.fit(target_predictors=selected_links, 
                   selected_targets=[target],
                   tau_max=tau_max) 
```

It would be nice if someone with more experience on this library (mine is limited) can comment if my code is ok or not.

**Reply by AGraciaSipan** [reply]
Hi @jvicentem but the `series_diff` argument you pass into the `selected_links_from_pcmci()` function you define where does it come from? I'm trying to understand your code.

**Comment by jakobrunge** [comment, marked as answer]
Rather try the CausalEffects class here. The Prediction class is not focused on contemporaneous predictions...
